modular_citadel/code/modules/arousal/toys/dildos.dm

```diff
@@ -131,3 +134,46 @@ obj/item/dildo/custom
 	random_color 		= TRUE
 	random_shape 		= TRUE
 	random_size 		= TRUE
+
+// Suicide acts, by request
+
+/obj/item/dildo/proc/manual_suicide(mob/living/user)
+	user.visible_message("<span class='suicide'>[user] finally finishes deepthroating the [src], and their life.</span>")
+	user.adjustOxyLoss(200)
+	user.death(0)
+
+/obj/item/dildo/suicide_act(mob/living/user)
+	user.visible_message("<span class='suicide'>[user] tears-up and gags as they shove [src] down their throat! Their face is turning blue! It looks like [user.p_theyre()] trying to commit suicide!</span>")
+	playsound(loc, 'sound/weapons/gagging.ogg', 50, 1, -1)
+	user.Stun(150)
+	user.adjust_blurriness(8)
+	user.adjust_eye_damage(10)
+	addtimer(CALLBACK(src, .proc/manual_suicide, user), 15)
```

Comment:
actually waitwaitwaitwait do_after() would be much more appropriate for this

Reply:
*deep breath*
alright here we go again